docs: update release instructions (#20210)
apache/superset: RELEASING/README.md

    svn update
```

> To minimize risk of mixing up your local development environment, it's recommended to work on the

Reviewer: ```suggestion
To minimize the risk of mixing up your local development environment, it's recommended to work on the
```

> release in a different directory than where the devenv is located. In this example we'll clone

Reviewer: ```suggestion
release in a different directory than where the devenv is located. In this example, we'll clone
```

the repo directly from the main `apache/superset` repo to a new directory `superset-release`:

```bash
cd <MY PROJECTS PATH>
git clone git@github.com:apache/superset.git superset-release
cd superset-release
```

We recommend setting up a virtual environment to isolate the python dependencies from your main
setup:

```bash
virtualenv venv
source venv/bin/activate
```

In addition, we recommend using the [`cherrytree`](https://pypi.org/project/cherrytree/) tool for

> automating cherrypicking, as it will help speed up the release process. To install `cherrytree`

Reviewer: ```suggestion
automating cherry picking, as it will help speed up the release process. To install `cherrytree`
```

This new branch will hold all PATCH and release candidates
that belong to the MAJOR.MINOR version.

### Creating an initial minor release (e.g. 1.5.0)

The MAJOR.MINOR branch is normally a "cut" from a specific point in time from the master branch.
When creating the initial minor release (e.g. 1.5.0), create a new branch:
```bash
git checkout master
git pull
git checkout -b ${SUPERSET_GITHUB_BRANCH}
git push origin $SUPERSET_GITHUB_BRANCH
```

Note that this initializes a new "release cut", and is NOT needed when creating a patch release
(e.g. 1.5.1).

### Creating a patch release (e.g. 1.5.1)

When getting ready to bake a patch release, simply checkout the relevant branch:

```bash
git checkout master
git pull
git checkout ${SUPERSET_GITHUB_BRANCH}
```

> ### Cherrypicking

Reviewer: ```suggestion
### Cherry picking
```

break by adding the `-error-mode break` flag (`-e break` for short):

```bash
cherrytree bake -r apache/superset -m master -l v${SUPERSET_GITHUB_BRANCH} -nd -e break ${SUPERSET_GITHUB_BRANCH}
```

After applying the cleanly merged cherries, `cherrytree` will specify the SHA of the conflicted cherry. To resolve the conflict,
simply issue the following command:

```bash
git cherry-pick <SHA>
```

Then fix all conflicts, followed by

```bash
git add -u  # add all changes
git cherry-pick --continue
```

After this, rerun all the above steps until all cherries have been picked, finally pushing all new commits to the release branch
on the main repo:

```bash
git push
```

### Updating changelog

Next, update the `CHANGELOG.md` with all the changes that are included in the release.
Make sure the branch has been pushed to `origin` to ensure the changelog generator
can pick up changes since the previous release.
Similar to `cherrytree`, the change log script requires a github token, either as an env var
(`GITHUB_TOKEN`) or as the parameter `--access_token`.

#### Initial release (e.g. 1.5.0)

When generating the changelog for an initial minor relese, you should compare with

> the previous release (in the example the previous release branch is `1.4`, so remember to

Reviewer: ```suggestion
the previous release (in the example, the previous release branch is `1.4`, so remember to
```

> update accordingly):

Reviewer: ```suggestion
update it accordingly):
```

python changelog.py --previous_version 0.37 --current_version 0.38 changelog --access_token {GITHUB_TOKEN} --risk
```

> The script will checkout both branches and compare all the PR's and output the line that are needed to be added to the

Reviewer: ```suggestion
The script will checkout both branches, compare all the PRs, and output the lines that are needed to be added to the
```

`CHANGELOG.md` file in the root of the repo. Remember to also make sure to update the branch id (with the above command
`1.5` needs to be changed to `1.5.0`)

Then, in `UPDATING.md`, a file that contains a list of notifications around
deprecations and upgrading-related topics,
make sure to move the content now under the `Next Version` section under a new
section for the new release.

#### Patch release (e.g. 1.5.1)

> To compare the forthcoming patch release with the latest release from the same branch, et the previous

Reviewer: "et the previous version"?

![et-o-extraterrestre-steven-spielberg-reproducao_widelg](https://user-images.githubusercontent.com/70410625/170694025-fbc88900-e882-4b05-8172-ee0407684816.jpeg)

Author: 🤣 🤣 🤣

Author: et the final version 😢 
![image](https://user-images.githubusercontent.com/33317356/170695795-b456aab7-9258-4d79-8f5a-a5bb70ea46fe.png)

Reviewer: 😆 😆 😆

> version as the tag of the previous release (in this example `1.5.0`; remember to update accordingly)

Reviewer: ```suggestion
version as the tag of the previous release (in this example `1.5.0`; remember to update it accordingly)
```
